plugins/inputs/t128_metrics/t128_metrics_test.go

```diff
@@ -0,0 +1,512 @@
+			},
+		},
+		ExpectedErrors: []string{
+			"status code 404 not OK for metric stats/404: it's not right",
+		},
+	},
+}
+
+var RequestFormationTestCases = []struct {
+	Name              string
+	ConfiguredMetrics []plugin.ConfiguredMetric
+	Endpoints         []Endpoint
+}{
+	{
+		Name: "empty request body with no parameters",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, "{}", "[]"}},
+	},
+	{
+		Name: "itemizes with no filter values for empty list",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{"interface": {}},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, `{"parameters": [{"name": "interface", "itemize": true}]}`, "[]"}},
+	},
+	{
+		Name: "includes parameter filter values",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{"interface": {"intf1", "intf2"}},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, `{
+				"parameters": [
+					{"name": "interface", "values": ["intf1", "intf2"], "itemize": true}
+				]
+			}`,
+			"[]"}},
+	},
+	{
+		Name: "includes multiple parameter filters",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{
+				"interface": {"intf1", "intf2"},
+				"node":      {"node1", "node2"},
+				"other":     {}},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, `{
+				"parameters": [
+					{"name": "interface", "values": ["intf1", "intf2"], "itemize": true},
+					{"name": "node", "values": ["node1", "node2"], "itemize": true},
+					{"name": "other", "itemize": true}
+				]
+			}`,
+			"[]"}},
+	},
+}
+
+func TestT128MetricsResponseProcessing(t *testing.T) {
+	for _, testCase := range ResponseProcessingTestCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			fakeServer := createTestServer(t, testCase.Endpoints)
+			defer fakeServer.Close()
+
+			plugin := &plugin.T128Metrics{
+				BaseURL:                 fakeServer.URL,
+				MaxSimultaneousRequests: 20,
+				ConfiguredMetrics:       testCase.ConfiguredMetrics,
+			}
+
+			var acc testutil.Accumulator
+			plugin.Init()
+
+			plugin.Gather(&acc)
+
+			var errorStrings []string = nil
+			for _, err := range acc.Errors {
+				errorStrings = append(errorStrings, err.Error())
+			}
+
+			require.ElementsMatch(t, testCase.ExpectedErrors, errorStrings)
+
+			// Timestamps aren't important, but need to match
+			for _, m := range acc.Metrics {
+				m.Time = time.Time{}
+			}
+
+			// Avoid specifying this unused type for each field
+			for _, m := range testCase.ExpectedMetrics {
+				m.Type = telegraf.Untyped
+			}
+
+			require.ElementsMatch(t, testCase.ExpectedMetrics, acc.Metrics)
```

Comment:
I'm assuming you are using this `require` module since it is used in other telegraf stuff? How does it compare to `gocmp`?

Reply:
I like it a lot! I don't think it has the flexibility to define custom comparisons for certain types, which means it probably wouldn't work for the complex metrics tests. But other than that, it's really nice.